### PR TITLE
fix: when updating building relation with previously coupled units

### PR DESCRIPTION
**Pull request by ArneD**

### CODE REVIEW

File: src/BuildingRegistry.Importer.Console/CommandGenerator.cs [hunk 1]
```diff
@@ -258,12 +258,33 @@ public IEnumerable<dynamic> GenerateUpdateCommandsFor(int key, DateTime @from, D
...
 
                 if (houseNumbersForUpdate.Any())
                 {
+                    var terrainObjectHouseNumberHouseNrCombinationsBeforeToday = importTerrainObjectHouseNumberCommands
+                        .Where(x => x.Timestamp <= from.ToCrabInstant())
+                        .Select(x => Tuple.Create(x.TerrainObjectHouseNumberId, x.HouseNumberId))
+                        .Distinct()
+                        .ToList();
+
+                    foreach (var terrainObjectHouseNumberFromCrab in importTerrainObjectHouseNumberFromCrabInRange.Where(x => !terrainObjectHouseNumberHouseNrCombinationsBeforeToday.Any(tuple => tuple.Item1 == x.TerrainObjectHouseNumberId && tuple.Item2 == x.HouseNumberId)))
```
Comment (ridingwolf) [inline src/BuildingRegistry.Importer.Console/CommandGenerator.cs:275]: extract that where clause from the foreach difinition. It's not a simple filter and makes it very hard to read what you are filtering.

File: src/BuildingRegistry.Importer.Console/CommandGenerator.cs [hunk 2]
```diff
@@ -258,12 +258,33 @@ public IEnumerable<dynamic> GenerateUpdateCommandsFor(int key, DateTime @from, D
...
+                    .Where(x => x.Timestamp > @from.ToCrabInstant() && x.Timestamp <= until.ToCrabInstant())
+                    .ToList();
+
+                var houseNumbersForUpdate = importTerrainObjectHouseNumberFromCrabInRange.Select(x => x.HouseNumberId).ToList();
 
                 if (houseNumbersForUpdate.Any())
                 {
+                    var terrainObjectHouseNumberHouseNrCombinationsBeforeToday = importTerrainObjectHouseNumberCommands
+                        .Where(x => x.Timestamp <= from.ToCrabInstant())
+                        .Select(x => Tuple.Create(x.TerrainObjectHouseNumberId, x.HouseNumberId))
```
Comment (ridingwolf) [inline src/BuildingRegistry.Importer.Console/CommandGenerator.cs:271]: using a Anonymous type will make the code far more readable then have to remember what item1 and item2 are

`.Select(x => new { x.TerrainObjectHouseNumberId, x.HouseNumberId })`

File: src/BuildingRegistry.Importer.Console/CommandGenerator.cs (hunk 1 above)
Comment (ridingwolf) [reply src/BuildingRegistry.Importer.Console/CommandGenerator.cs:275]: use meaningful names in the lambda
`x, tuple, item1, item2` make it very hard to read what's being compared

File: src/BuildingRegistry.Importer.Console/CommandGenerator.cs [hunk 3]
```diff
@@ -258,12 +258,33 @@ public IEnumerable<dynamic> GenerateUpdateCommandsFor(int key, DateTime @from, D
...
+
+                    foreach (var terrainObjectHouseNumberFromCrab in importTerrainObjectHouseNumberFromCrabInRange.Where(x => !terrainObjectHouseNumberHouseNrCombinationsBeforeToday.Any(tuple => tuple.Item1 == x.TerrainObjectHouseNumberId && tuple.Item2 == x.HouseNumberId)))
+                    {
+                        var allNewSubaddressIds = importSubaddressCommands.Where(subaddressFromCrab => subaddressFromCrab.HouseNumberId == terrainObjectHouseNumberFromCrab.HouseNumberId).Select(x => x.SubaddressId);
+                        foreach (var newSubaddressId in allNewSubaddressIds)
+                        {
+                            allCommands = allCommands.Concat(allTerrainObjectCommands
+                                    .Except(allCommands)
+                                    .Where(x =>
+                                        (x.Item1 is ImportSubaddressFromCrab importSubaddressFromCrab && importSubaddressFromCrab.SubaddressId == newSubaddressId && importSubaddressFromCrab.TerrainObjectHouseNumberId == terrainObjectHouseNumberFromCrab.TerrainObjectHouseNumberId)))
```
Comment (ridingwolf) [inline src/BuildingRegistry.Importer.Console/CommandGenerator.cs:283]: split into multiples lines for readability

File: src/BuildingRegistry.Importer.Console/CommandGenerator.cs (hunk 1 above)
Comment (ArneD) [reply src/BuildingRegistry.Importer.Console/CommandGenerator.cs:275]: Can't extract where clause when using anonymous objects. I'll reformat to make it more readable

**Pull request merged**